// @elsikora/ecorpay-types/merchant/ways/ways.t.ts
export interface IMerchantFee {
  external: number;
  internal: number;
  depositRatio?: number;
  transferType?: boolean;
}

export type TPurseType = 'wallet' | 'card';
export type TCurrencyType = 'crypto' | 'fiat';

interface IWayType {
  id: string;
  name: string;
  pursePattern: string;
  purseType: TPurseType;
  mask: string;
  createdAt?: Date;
  updatedAt?: Date;
}

export interface IWayCurrency {
  id: string;
  code: string;
  type: TCurrencyType;
  precision: number;
  createdAt?: Date;
  updatedAt?: Date;
}

export interface IResponse {
  id: string;
  type: IWayType;
  deposit: boolean;
  currency: IWayCurrency;
  transfer: boolean;
  fee: IMerchantFee;
  enabled: boolean;
  limits: {
    deposit: {
      min: number;
      max: number;
    },
    transfer: {
      min: number;
      max: number;
    }
  };
  createdAt: Date;
  updatedAt: Date;
};


//deposit list 
export interface IMerchantWay extends IResponse {
  hide: {
    deposit: number;
    transfer: number;
    activity: number;
  };
}




/*
export interface IResponse extends IResponseCreate {
  balance: IMerchantBalance;
  ways: IResponseFindById;
  callback: IMerchantCallbacks;
  key: string;
  secret: string;
  updatedAt: Date;
}
*/
