// @elsikora/ecorpay-types/merchant/transfer/transfer.t.ts
import { IPaginationList } from "../../shared/shared.t";
import { IMerchantWay, IResponse as IWay } from "../ways/ways.t";

export enum ETransferStatus {
	NEW = "new",
	WAITING = "waiting",
	PROCESS = "process",
	PAID = "paid",
	FAILED = "failed",
};

export enum ETransferType {
	EXTERNAL = "external",
	INTERNAL = "internal",
};

export interface ICreate {
  way: string;
  amount: number;
  internalID: string;
  customerID: string;
  purse: '',
  description: string;
  ip: string;
};

export interface IPaginationFilters extends IPaginationList {
  description: string;
  way: string;
  currency: string;
  status: ETransferStatus[];
  amountFrom: number;
  amountTo: number;
  dateFrom: string;
  dateTo: string;
};

export interface IFindById {
  id: string;
};

type OmittedIMerchantWay = Omit<IMerchantWay, "deposit" | "transfer" | "fee" | "limits">;

export interface IResponse {
  id: string;
  amount: number;
  way: OmittedIMerchantWay;
  merchant: { name: string };
  description: string;
  status: ETransferStatus;
  purse: string;
  type: ETransferType;
  fee: { internal: number; transferType: boolean; }
  createdAt: Date;
  updatedAt: Date;
};
